### PR TITLE
WORK:3387747 Added new Brooklyn Co-existance feature Powershell commandlets

### PR DESCRIPTION
**Pull request by Nilambari**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Nilambari__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (nilamd). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by Nilambari** [comment]
@stankovski As per your suggestion, I had raised PR request against branch: release-0.9.2 in the afternoon.  Can you Please help accept this PR? Thanks a lot!

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureLocalNetworkGateway.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Gateway
+{
+    [Cmdlet(VerbsCommon.Get, "AzureLocalNetworkGateway"), OutputType(typeof(GetLocalNetworkGatewayContext))]
+    public class GetAzureLocalNetworkGateway : NetworkCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = false, HelpMessage = "Local network gateway Id.")]
+        [ValidateGuid]
+        [ValidateNotNullOrEmpty]
+        public string gatewayId
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureLocalNetworkGateway.cs]: Incorrect name casing

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGateway.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Gateway
+{
+    [Cmdlet(VerbsCommon.Get, "AzureVirtualNetworkGateway"), OutputType(typeof(GetVirtualNetworkGatewayContext))]
+    public class GetAzureVirtualNetworkGateway : NetworkCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = false, HelpMessage = "Virtual network gateway Id.")]
+        [ValidateGuid]
+        [ValidateNotNullOrEmpty]
+        public string gatewayId
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGateway.cs]: Same as above

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Gateway
+{
+    [Cmdlet(VerbsCommon.Get, "AzureVirtualNetworkGatewayConnection"), OutputType(typeof(GetVirtualNetworkGatewayConnectionContext))]
+    public class GetAzureVirtualNetworkConnectionGateway : NetworkCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = false, HelpMessage = "Virtual network gateway Id.")]
+        [ValidateGuid]
+        [ValidateNotNullOrEmpty]
+        public string gatewayId
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs]: Casing

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            get;
+            set;
+        }
+
+        [Parameter(Position = 1, Mandatory = false, HelpMessage = "Virtual network gateway Connected entityId.")]
+        [ValidateGuid]
+        [ValidateNotNullOrEmpty]
+        public string connectedentityId
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs]: Casing

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Gateway
+{
+    [Cmdlet(VerbsCommon.Get, "AzureVirtualNetworkGatewayDiagnostics"), OutputType(typeof(ManagementOperationContext))]
+    public class GetAzureVirtualNetworkGatewayDiagnosticsV2 : NetworkCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "The virtual network gateway id.")]
+        [ValidateGuid]
+        [ValidateNotNullOrEmpty]
+        public string gatewayId
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs]: Casing

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+// ----------------------------------------------------------------------------------
+
+using Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Gateway.Model;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Gateway
+{
+    [Cmdlet(VerbsCommon.Get, "AzureVirtualNetworkGatewayDiagnostics"), OutputType(typeof(ManagementOperationContext))]
+    public class GetAzureVirtualNetworkGatewayDiagnosticsV2 : NetworkCmdletBase
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs]: Why is there a V2 in the name?

File: src/ServiceManagement/Network/Commands.Network/Gateway/Model/GetVirtualNetworkGatewayConnectionContext.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Network
+{
+    using Microsoft.WindowsAzure.Management.Network.Models;
```
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/Model/GetVirtualNetworkGatewayConnectionContext.cs]: All using statements should be above namespce declaration

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureLocalNetworkGateway.cs (hunk 1 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureLocalNetworkGateway.cs]: Done.

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs (hunk 6 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs]: These are new version of APIs similar to existing ones. and currently we support both the versions, so added V2 in the name of new APIs for user's understanding.

File: src/ServiceManagement/Network/Commands.Network/Gateway/Model/GetVirtualNetworkGatewayConnectionContext.cs (hunk 7 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/Model/GetVirtualNetworkGatewayConnectionContext.cs]: Done.

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs (hunk 6 above)
Comment (stankovski) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs]: User doesn't see the class name and the cmdlet does not reflect the difference.

**Comment by stankovski** [comment]
Add tests

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGateway.cs (hunk 2 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGateway.cs]: Done.

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs (hunk 3 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs]: Done.

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs (hunk 4 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayConnection.cs]: Done.

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs (hunk 5 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs]: Done.

**Comment by Nilambari** [comment]
@stankovski We do not have tests for existing commandlets as well. I have added tests in Sdk-for-net repo for all these new APIs. I have validated all the commandlets work fine. I can definitely work on adding tests for PS cmdlets, Thanks a lot for the review!

File: src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs (hunk 6 above)
Comment (Nilambari) [inline src/ServiceManagement/Network/Commands.Network/Gateway/GetAzureVirtualNetworkGatewayDiagnosticsV2.cs]: Agree. Removed V2 from class names too.

**Comment by hovsepm** [comment]
Please open PR against Dev branch.

**Pull request closed without merging**